src/audit_log/audit_log.cc

```diff
@@ -372,6 +372,16 @@ bool AuditLog::merge(AuditLog *from, std::string *error) {
     return init(error);
 }
 
+bool AuditLog::reopen(std::string *error) {
+    //dd("audit log reopen");
+    if (m_writer) {
+        //dd("got a writer");
+        return m_writer->reopen(error);
```

Comment:
Since AuditLog already owns the two possible log paths, we can probably stop involving the writer interface and directly ask SharedFiles to reopen here instead.

Reply:
Done.